compiler/ast/errorhandling.nim

```diff
@@ -171,10 +175,10 @@ proc buildErrorList(config: ConfigRef, n: PNode, errs: var seq[PNode]) =
   of nkEmpty .. nkNilLit:
     discard
   of nkError:
-    errs.add n
     buildErrorList(config, n[wrongNodePos], errs)
+    errs.add n
   else:
-    for i in countdown(n.len - 1, 0):
+    for i in 0..<n.len:
```

Comment:
I personally would put explanation comment about why exactly iteration is done this way and not the other. Of course this is not a blocker, but given this is one of the most important procs that defines the whole logic behind error walker, it would still be nice to see more explanations.